ykcapi/yk.h

```diff
@@ -48,4 +48,42 @@ YkLocation yk_location_new(void);
 // will occur.
 void yk_location_drop(YkLocation);
 
+void yk_stopgap(void *addr, uintptr_t size, uintptr_t retaddr, void *rsp);
+
+void __llvm_deoptimize(void* addr, uintptr_t size) {
+    // Push all registers to the stack before they can be clobbered, so that we
+    // can find their values after parsing in the stackmap. After that we store
+    // the value of RSP in a variable so we know where to look for the register
+    // values. The order in which we push the registers is equivalent to the
+    // Sys-V x86_64 ABI, which the stackmap format uses as well. However, since
+    // we are interested in the value of RSP in the previous stack frame, which
+    // has changed when we entered the __llvm_deoptimize function, we push RBP
+    // instead, which is equivalent. Note, that because of this there is no
+    // (easy) way to access RBP from the previous frame, so we just assume it's
+    // never referenced by the stackmap.
+    void *rsp;
+    __asm__ volatile (
```

Comment:
I think this block of code answers my question about what registers are spilled.

Should we move this block into a method in `Registers` so that they are right next to each other?

Reply:
Not sure we can, to be honest. Don't know if calling another function here would mess with the register values. The very first thing we need to do in `llvm_deoptimize` is save the registers so that they don't get overwritten.

Reply:
I see. At least a comment saying that this must stay in parity with `Registers` then?

Reply:
I think this is already addressed in efcb294 via the added `unreachable!`.